### PR TITLE
ReplaceOpaqueTypesWithUnderlyingTypes: Use resilience expansion instead of inlinable attribute

### PR DESCRIPTION
**Pull request by aschwaighofer**

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test linux

File: lib/AST/Type.cpp [hunk 1]
```diff
@@ -2484,9 +2484,18 @@ bool ReplaceOpaqueTypesWithUnderlyingTypes::shouldPerformSubstitution(
 
   // Allow replacement of opaque result types of inlineable function regardless
   // of resilience and in which context.
-  if (namingDecl->getAttrs().hasAttribute<InlinableAttr>()) {
-    return true;
+  if (auto *afd = dyn_cast<AbstractFunctionDecl>(namingDecl)) {
```
Comment (slavapestov) [inline lib/AST/Type.cpp]: Can you add some tests where either the property itself or just the getter is inlinable?

Comment (aschwaighofer) [reply lib/AST/Type.cpp]: There is one. It failed when I had not added the code for abstract storage decl.

https://github.com/apple/swift/blob/master/test/SILOptimizer/specialize_opaque_type_archetypes.swift#L335

https://github.com/apple/swift/blob/master/test/SILOptimizer/Inputs/specialize_opaque_type_archetypes_3.swift#L59

Comment (slavapestov) [reply lib/AST/Type.cpp]: Ah ok. A case you might want to test that the old logic definitely would have missed is an `@_alwaysEmitIntoClient` decl. These are the same as inlinable except they have different linkage -- but don't have an InlinableAttr.

**Comment by aschwaighofer** [comment]
@swit-ci Please test

**Comment by aschwaighofer** [comment]
@swift-ci Please test

**Pull request merged**